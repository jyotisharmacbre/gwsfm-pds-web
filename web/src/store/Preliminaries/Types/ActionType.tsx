export enum ActionType {
	PRELIMINARY_ADD_SUCCESS = 'PRELIMINARY_ADD_SUCCESS',
	PRELIMINARY_ADD_ERROR = ' PRELIMINARY_ADD_ERROR',
	PRELIMINARY_EDIT_SUCCESS = ' PRELIMINARY_EDIT_SUCCESS',
	PRELIMINARY_EDIT_ERROR = ' PRELIMINARY_EDIT_ERROR',
	GET_PRELIMINARY_SUCCESS = 'GET_PRELIMINARY_SUCCESS',
	GET_PRELIMINARY_ERROR = 'GET_PRELIMINARY_ERROR',
	UPDATE_PRELIMINARY_STATE = 'UPDATE_PRELIMINARY_STATE',
	RESET_PRELIMINARY_STATE = 'RESET_PRELIMINARY_STATE',
	SET_LOADING_TRUE = 'SET_LOADING_TRUE'
}
