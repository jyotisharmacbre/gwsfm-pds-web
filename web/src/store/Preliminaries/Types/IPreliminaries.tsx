export interface IPreliminaries {
PreliminaryId :string;
ProjectId :string;
PreliminariesItemId :string;
PreliminariesComponentId :string;
NameOfSupplier :string;
NoOfHours :number;
HourRate :number;
TotalCost :number;
GrossMargin :number;
Comments :string;
  }
  