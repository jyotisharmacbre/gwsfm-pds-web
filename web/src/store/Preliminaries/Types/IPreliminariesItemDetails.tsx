export interface IPreliminariesItemDetails {
  nameOfSupplier: string;
  noOfHours: number;
  hourRate: number;
  totalCost: number;
  grossMargin: number;
  comments: string;
}
