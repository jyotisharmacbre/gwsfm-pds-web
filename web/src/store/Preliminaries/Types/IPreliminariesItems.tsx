export interface IPreliminariesItems {
  itemId: string;
  itemName: string;
  nameOfSupplier: string;
  noOfHours: number;
  hourRate: number;
  totalCost: number;
  grossMargin: number;
  comments: string;
  preliminaryId:string;
}
