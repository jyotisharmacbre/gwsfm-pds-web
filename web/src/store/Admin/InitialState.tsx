import { IAdminState } from './Types/IAdminState';
import { IAdminDefaults } from './Types/IAdminDefault';

export const initialState: IAdminState = {
	adminDefaultValues: [],
	error: ''
};
