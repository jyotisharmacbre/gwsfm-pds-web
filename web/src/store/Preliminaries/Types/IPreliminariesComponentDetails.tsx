import { IPreliminariesItems } from './IPreliminariesItems';
export interface IPreliminariesComponentDetails {
  componentId: string;
  componentName: string;
  items: Array<IPreliminariesItems>;
}
