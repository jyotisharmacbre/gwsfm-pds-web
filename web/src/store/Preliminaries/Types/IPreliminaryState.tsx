import { IPreliminariesComponentDetails } from './IPreliminariesComponentDetails';
import Notify from '../../../enums/Notify';

export interface IPreliminaryForm {
  preliminaryDetails: Array<IPreliminariesComponentDetails>;
 
}

export interface IPreliminaryState {
  projectId: string;
  preliminaryDetails:Array<IPreliminariesComponentDetails>;
}
