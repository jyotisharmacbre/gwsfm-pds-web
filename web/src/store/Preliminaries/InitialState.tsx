import Notify from '../../enums/Notify';
import { IPreliminaryState } from './Types/IPreliminaryState';
import EventType from '../../enums/EventType';
export const initialState: IPreliminaryState = {
    preliminaryDetails: [],
    notify: Notify.none,
    event: EventType.none
  };