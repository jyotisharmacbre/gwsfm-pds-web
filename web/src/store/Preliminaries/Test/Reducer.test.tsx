import {ActionType} from '../Types/ActionType';
import preliminaryReducer from '../Reducer';
import { initialState } from '../InitialState';
import {lookupData} from "./ReducerTestData";
import nock from 'nock';
import { baseURL } from '../../../client/client';

nock(baseURL)
.get('/api/Preliminaries/4d27e2e1-843d-435a-b27c-03dca70ce232/preliminaryDetails')
.reply(200, "Preliminary data fetched successfully");

nock(baseURL)
.put('/api/Preliminaries/preliminaryDetails')
.reply(200, "Preliminary data update successfully");

nock(baseURL)
.post('/api/Preliminaries/preliminaryDetails')
.reply(200, "Preliminary data saved successfully");

describe('preliminaryReducer Reducer', () => {
 
  it('should fetch preliminary data', () => {
    const getLookupDataAction: any = {
      type: ActionType.GET_PRELIMINARY_SUCCESS
    };
    expect(
        preliminaryReducer(initialState, getLookupDataAction)
    ).toMatchSnapshot();
  });
  it('should populate error.If get fails', () => {
    const dataAction: any = {
      type: ActionType.GET_PRELIMINARY_ERROR
    };
    expect(
        preliminaryReducer(initialState, dataAction)
    ).toMatchSnapshot();
  });
    it('should update preliminary data', () => {
      const dataAction: any = {
        type: ActionType.PRELIMINARY_EDIT_SUCCESS
      };
      expect(
          preliminaryReducer(initialState, dataAction)
      ).toMatchSnapshot();
    });
    it('should populate error.If update fails', () => {
      const dataAction: any = {
        type: ActionType.PRELIMINARY_EDIT_ERROR
      };
      expect(
          preliminaryReducer(initialState, dataAction)
      ).toMatchSnapshot();
    });
    it('should save preliminary data', () => {
      const dataAction: any = {
        type: ActionType.PRELIMINARY_ADD_SUCCESS
      };
      expect(
          preliminaryReducer(initialState, dataAction)
      ).toMatchSnapshot();
    });
    it('should populate error.If save fails', () => {
      const dataAction: any = {
        type: ActionType.PRELIMINARY_ADD_ERROR
      };
      expect(
          preliminaryReducer(initialState, dataAction)
      ).toMatchSnapshot();
    });
  });
      
