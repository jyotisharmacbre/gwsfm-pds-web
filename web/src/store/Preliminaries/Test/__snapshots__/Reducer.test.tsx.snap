// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preliminaryReducer Reducer should fetch preliminary data 1`] = `
Object {
  "notify": 0,
  "preliminaryDetails": undefined,
}
`;

exports[`preliminaryReducer Reducer should populate error.If get fails 1`] = `
Object {
  "notify": 0,
  "preliminaryDetails": Array [],
}
`;

exports[`preliminaryReducer Reducer should populate error.If save fails 1`] = `
Object {
  "notify": 2,
  "preliminaryDetails": Array [],
}
`;

exports[`preliminaryReducer Reducer should populate error.If update fails 1`] = `
Object {
  "notify": 0,
  "preliminaryDetails": Array [],
}
`;

exports[`preliminaryReducer Reducer should save preliminary data 1`] = `
Object {
  "notify": 1,
  "preliminaryDetails": Array [],
}
`;

exports[`preliminaryReducer Reducer should update preliminary data 1`] = `
Object {
  "notify": 1,
  "preliminaryDetails": Array [],
}
`;
